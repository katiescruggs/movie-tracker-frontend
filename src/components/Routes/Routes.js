import React from 'react';
import { Route } from 'react-router-dom';
import CardContainer from '../../containers/CardContainer/CardContainer';
import Login from '../../containers/Login/Login';

const Routes = () => {
  return (
    <div className="routes">
        <Route 
          exact path='/login'  
          render={props => <Login {...props} />}
        />
        <Route 
          exact path='/register'  
          render={props => <Login {...props} showRegister />}
        />
        <Route path='/' component={CardContainer} />
    </div>
  );
}

export default Routes;