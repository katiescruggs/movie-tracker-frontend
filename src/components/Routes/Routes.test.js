import React from 'react';
import { shallow } from 'enzyme';
import Routes from './Routes';

describe('Routes tests', () => {
  let routes;

  beforeEach(() => {
    routes = shallow(<Routes />);
  });

  it('should be defined', () => {
    expect(routes).toBeDefined();
  });

  it('should match the snapshot', () => {
    expect(routes).toMatchSnapshot();
  });
});