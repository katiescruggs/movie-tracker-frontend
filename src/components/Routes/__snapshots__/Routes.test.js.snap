// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Routes />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Route
          exact={true}
          path="/login"
          render={[Function]}
/>,
        <Route
          exact={true}
          path="/register"
          render={[Function]}
/>,
        <FilmReel />,
        <Route
          component={[Function]}
          path="/"
/>,
      ],
      "className": "routes",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/login",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/register",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Route
            exact={true}
            path="/login"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/register"
            render={[Function]}
/>,
          <FilmReel />,
          <Route
            component={[Function]}
            path="/"
/>,
        ],
        "className": "routes",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/login",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/register",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
